src/python/pants/backend/python/util_rules/lockfile_metadata.py

```diff
@@ -53,36 +42,46 @@ def from_lockfile(cls, lockfile: bytes) -> LockfileMetadata:
             elif in_metadata_block:
                 metadata_lines.append(line[2:])
 
-        error_suffix = "To resolve this error, you will need to regenerate the lockfile."
+        error_suffix = "To resolve this error, you will need to regenerate the lockfile by running `./pants generate-lockfiles"
+
+        if lockfile_scope_name:
+            error_suffix += "--resolve={tool_name}"
+
+        error_suffix += "`."
+
+        lockfile_description: str = ""
+        if lockfile_scope_name:
+            lockfile_description = f"the lockfile for `{lockfile_scope_name}`"
+        else:
+            """this lockfile."""
```

Comment:
This doesn't do anything. Also prefer ternary expressions: https://www.pantsbuild.org/docs/style-guide#prefer-conditional-expressions-ternary-expressions

Reply:
ack

Reply:
Thanks for catching the typo

Reply:
This is now a 4-level conditional, and it reads pretty poorly as a ternary expression, so I've kept it as an if statement for now. Let me know if this is a problem.